### PR TITLE
HOV-14 | Checkbox Component

### LINKED ISSUE #39: HOV-14 | Checkbox Component
## Expected Behavior
Make a `<Checkbox>` Component
Features:
- Should come with default `input` props
- Inherit from `<input type="checkbox">`

## Current Behavior
No checkbox component

- [ ] Make spec for modal component
- [ ] Make the base component with API and without styles
- [ ] add styling

### PR DESCRIPTION
**Pull request by Raalzz**
## Description

Adds Checkbox Component

## Motivation and Context

Closes #39 

## How Has This Been Tested?

Tested in local and Docs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pruthvi2103** [comment]
is this ready for review @Raalzz ?

**Comment by Raalzz** [comment]
Yes

File: lib/src/components/_internal/Icons/SvgCheck.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      <path
+        fill="currentColor"
+        stroke="none"
+        d="M11.915 1.548 10.367 0 4.045 6.315 1.557 3.827 0 5.373l4.045 4.046 7.87-7.871Z"
+      ></path>
+    </svg>
+  );
+};
+
+export { SvgCheck };
```
Comment (pruthvi2103) [inline lib/src/components/_internal/Icons/SvgCheck.tsx:13]: Use the `Icon` Component to wrap this

File: lib/src/components/_internal/Icons/SvgDash.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const SvgDash = () => {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 32 6">
```
Comment (pruthvi2103) [inline lib/src/components/_internal/Icons/SvgDash.tsx:3]: use `Icon` instead of `svg`

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
Added comments

File: lib/src/components/_internal/Icons/SvgDash.tsx (hunk 2 above)
Comment (Raalzz) [reply lib/src/components/_internal/Icons/SvgDash.tsx]: Done

**Review by pruthvi2103 — APPROVED** [review]
LGTM

**Pull request merged**